packages/pages/src/dev/client/entry-client.tsx

```diff
@@ -36,15 +36,18 @@ const hydrate = async () => {
     },
   };
 
-  const { default: Component } = await template.getComponent();
+  const { default: Component, render } = await template.getComponent();
 
-  if (!Component) {
-    console.error("Default export missing in template: " + template.path);
+  if (!Component && !render) {
+    console.error(
+      "Either a default export or render function is required in template: " +
+        template.path
+    );
     return;
   }
 
   ReactDOM.hydrate(
```

Comment:
`render` doesn't return a React.Element. It doesn't need to be hydrated. It is supposed to represent the whole page.

Reply:
I don't think anything needs to be rendered client-side